### PR TITLE
Update ImportOCAF2.cpp

### PR DESCRIPTION
**Pull request by mosfet80**
ImportOCAF2 code optimizations.

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [x]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the issue ID number from our [Issues database](https://github.com/FreeCAD/FreeCAD/issues) in case a particular commit solves or is related to an existing issue. Ex: `Draft: fix typos - fixes #4805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by mosfet80** [comment]
applyFaceColors(feature,{info.faceColor}); is rerun    if(hasFaceColors)
    applyEdgeColors(feature,{info.edgeColor}); is rerun   if(hasEdgeColors)

**Comment by noahwilliamsson** [comment]
LGTM.

Nit: please consider improving your commit messages to make things easier for reviewers or those looking at the git history.  For example: `Import: ImportOCAF2: dead code removal, formatting`.

By prefixing MRs with the module name, it's easier for others to quickly identify PRs related to areas they care about on https://github.com/FreeCAD/FreeCAD/pulls or in apps such as https://devhubapp.com (GitHub activity tracker for repos etc).

If you're using Git from the command line, updating the commit message of the most recent commit is as easy as:

```
git commit --amend
```

To update a commit message further back in the history, try an interactive rebase with the last N commits.  For example:

```
git rebase -i HEAD~5
```
In the editor that appears, you can change `pick` to `r` (`reword`) and save the file.  Git will then allow you to edit the commit message of that commit, before continuing to rebase the history.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6643">pipeline status</a> for feature branch [PR_6643](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_6643). Pipeline [498534584](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/498534584) was triggered at [82a7bcd](https://github.com/FreeCAD/FreeCAD/pull/6643/commits/82a7bcd1f495575c09dc96697487ddc79fb5e69c). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by yorikvanhavre** [comment]
Looks good to me too, but there might be a reason why the code block under the `#if 0` is still there... @realthunder I think you wrote it?

File: src/Mod/Import/App/ImportOCAF2.cpp [hunk 1]
```diff
@@ -305,22 +305,7 @@ App::DocumentObject *ImportOCAF2::expandShape(
...
-    auto baseShape = shape.Located(TopLoc_Location());
-    auto it = myShapes.find(baseShape);
-    if(it!=myShapes.end()) {
-        auto link = static_cast<App::Link*>(doc->addObject("App::Link","Link"));
-        link->Visibility.setValue(false);
-        link->setLink(-1,it->second.obj);
-        setPlacement(&link->Placement,shape);
-        return link;
-    }
-#endif
```
Comment (realthunder) [inline src/Mod/Import/App/ImportOCAF2.cpp:323]: The comment is my understanding by testing and browsing through OCC code, which is kind of a surprise to me. I am not sure if OCC will change this behavior in the future. I think it worth keeping the comment and code here.

File: src/Mod/Import/App/ImportOCAF2.cpp [hunk 2]
```diff
@@ -450,12 +435,24 @@ bool ImportOCAF2::createObject(App::Document *doc, TDF_Label label,
         feature->Shape.setValue(shape);
         // feature->Visibility.setValue(false);
     }
-    applyFaceColors(feature,{info.faceColor});
-    applyEdgeColors(feature,{info.edgeColor});
```
Comment (realthunder) [inline src/Mod/Import/App/ImportOCAF2.cpp:454]: Please check out [here](https://github.com/FreeCAD/FreeCAD/blob/950a569b44c5220a456c49e0455ba54045c03288/src/Mod/Import/Gui/AppImportGuiPy.cpp#L309) for different treatment of array with single and multiple elements.

**Comment by donovaly** [comment]
@yorikvanhavre , can you please have a look and decide if this goes in for 0.20 or 0.21?

**Comment by donovaly** [comment]
@mosfet80 , @yorikvanhavre  - Ping!

**Comment by yorikvanhavre** [comment]
i think per @realthunder 's comment it's best to keep the commented code in, the way it is now. Can you adapt your PR, @mosfet80 ?

**Pull request closed without merging**